plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php

```diff
@@ -375,6 +376,55 @@ function ( $all, $curr ) {
 		return null;
 	}
 
+	/**
+	 * Return number of setup tasks remaining
+	 *
+	 * @return number
+	 */
+	public static function setup_tasks_remaining () {
+		if( ! Features::is_enabled( 'tasklist-setup-experiment-1' )) {
+			return;
+		}
+
+		$setup_list = self::get_list( 'setup_experiment_1' );
+
+		if( $setup_list->is_hidden() || $setup_list->is_complete() ) {
+			return;
+		}
+
+		$remaining_tasks = array_values(
+			array_filter(
+				$setup_list->get_viewable_tasks(),
+				function( $task ) {
+					return ! $task->is_complete();
+				}
+			)
+		);
+
+		return count( $remaining_tasks );
+	}
+
+	/**
+	 * Add badge to homescreen menu item for remaining tasks
+	 */
+	public static function menu_task_count() {
+		global $submenu;
+
+		$tasks_count = self::setup_tasks_remaining();
+
+		if( !$tasks_count ) {
+			return;
+		}
+
+		foreach ( $submenu['woocommerce'] as $key => $menu_item ) {
```

Comment:
This is using basically the same logic and styles as the preexisting orders badge, but I'm receptive to a more elegant way if one exists.